### PR TITLE
Use external for nested comments

### PR DESCRIPTION
**Pull request by 314eter**
It seems impossible to parse nested comments internally because `extras` accepts single tokens only.

### CODE REVIEW

File: src/scanner.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+void tree_sitter_external_extra_tokens_external_scanner_reset(void *payload) {}
+
+unsigned tree_sitter_ocaml_external_scanner_serialize(void *payload, char *buffer) {
+  return true;
+}
+
+void tree_sitter_ocaml_external_scanner_deserialize(void *payload, const char *buffer, unsigned length) {}
+
+bool tree_sitter_ocaml_external_scanner_scan(void *payload, TSLexer *lexer, const bool *whitelist) {
+  int *depth = (int*) payload;
```
Comment (maxbrunsfeld) [inline src/scanner.c]: Could this just be `int depth = 0;`? I would think we wouldn't need to carry any state between calls to `scan`.

File: src/scanner.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#include <tree_sitter/parser.h>
+
+enum {
+  COMMENT
+};
+
+void *tree_sitter_ocaml_external_scanner_create() {
+  return malloc(sizeof(int));
```
Comment (maxbrunsfeld) [inline src/scanner.c]: Could this just return `NULL`?

File: src/scanner.c (hunk 1 above)
Comment (314eter) [reply src/scanner.c]: That's right, I was still trying to figure out how external works when I wrote this.

**Comment by maxbrunsfeld** [comment]
Thanks so much! Great PR.

**Pull request merged**